src/software/backend/simulation/physics/physics_robot.cpp

```diff
@@ -1,16 +1,55 @@
 #include "software/backend/simulation/physics/physics_robot.h"
 
-#include "shared/constants.h"
+#include <algorithm>
+#include <numeric>
+
 #include "software/backend/simulation/physics/box2d_util.h"
+#include "software/backend/simulation/physics/physics_object_user_data.h"
+extern "C"
+{
+#include "firmware/main/shared/physics.h"
+}
 
-PhysicsRobot::PhysicsRobot(std::shared_ptr<b2World> world, const Robot& robot)
+// We are only allowed to cover a fraction of the ball according to the rules, so we use
+// this to calculate the depth. We assume the ball can be dribbled as long as it is
+// anywhere within this small area.
+const double PhysicsRobot::dribbler_depth =
+    BALL_MAX_RADIUS_METERS * 2 * MAX_FRACTION_OF_BALL_COVERED_BY_ROBOT;
+// We can use a very small value for the chicker thickness since the
+// chicker just needs to be large enough to collide with the ball and detect collisions
+// without letting the ball tunnel and collide with the robot body
+const double PhysicsRobot::chicker_thickness = 0.005;
```

Comment:
nit: can we put this number in the header file

Reply:
Unfortunately the only one I can put the in the header file is the `chicker_thickness`. The only way to initialize a static member in the header is if it is a `constexpr`. For whatever reason, `#define` variables can't be used to make a `constexpr` which means I can't put `dribbler_depth` or `total_chicker_depth` in the header.

So since I could only move 1/3, I think it's better to leave for consistency.